### PR TITLE
Fixes and tests `Field(u::Field; indices)` and `Field(u::AbstractOperation; indices)`

### PR DESCRIPTION
**Pull request by tomchor**
Closes https://github.com/CliMA/Oceananigans.jl/issues/2391

### CODE REVIEW

**Comment by glwagner** [comment]
Mind if I commit to this PR?

File: test/test_field.jl [hunk 1]
```diff
@@ -218,6 +218,8 @@ end
                     field_sz = test_field_sizes[t]
                     parent_sz = test_parent_sizes[t]
                     f_view = view(f, indices...)
+                    f_sliced = Field(f, indices)
+                    f2_sliced = Field(f^2, indices)
```
Comment (glwagner) [inline test/test_field.jl:222]: There's no other test for abstract operations in this file... should we put this in `test_abstract_operations.jl`? The other test is perfect.

Comment (glwagner) [reply test/test_field.jl:222]: Or actually I think `test_computed_field.jl` is the ideal place to test this.

Comment (tomchor) [reply test/test_field.jl:222]: Yeah, move those around however you think is necessary. I never know what the proper place for tests are; that was just by best guess

**Comment by tomchor** [comment]
> Mind if I commit to this PR?

Commit away!

File: test/test_field.jl (hunk 1 above)
Comment (glwagner) [reply test/test_field.jl]: It's subjective!

**Review by glwagner — APPROVED** [review]
Hopefully that does it. We were missing 1) `validate_indices` inside the computed `Field` constructor and 2) a translator from `Field(f::Field; indices) = view(f, indices...)`.

File: src/Fields/field.jl [hunk 2]
```diff
@@ -157,7 +157,7 @@ function Field(loc::Tuple,
 end
     
 Field(z::ZeroField; kw...) = z
-Field(f::Field) = f # hmm...
+Field(f::Field; indices=f.indices) = view(f, indices...) # hmm...
```
Comment (tomchor) [inline src/Fields/field.jl:160]: :+1: this is probably all that was needed to fix the bug

Comment (tomchor) [reply src/Fields/field.jl:160]:
```suggestion
Field(f::Field; indices=f.indices) = view(f, indices...)
```

Comment (glwagner) [reply src/Fields/field.jl:160]: There's still a few uncertain things here, like whether we should respect / include other keyword arguments (boundary conditions, data, etc...)

Comment (tomchor) [reply src/Fields/field.jl:160]: Then can we include a more descriptive comment please?

Also, doesn't removing the line `Field(f::Field) = f` mean that `Field(u)` won't work anymore? Maybe we wanna keep that line?

Comment (glwagner) [reply src/Fields/field.jl:160]: `Field(u)` works; the kwarg is default `u.indices` (eg the same indices)

**Pull request merged**